pallets/automation-time/src/lib.rs

```diff
@@ -245,6 +285,54 @@ pub mod pallet {
 			Ok(().into())
 		}
 
+		/// Schedule a task to fire an event with a custom message.
+		///
+		/// Before the task can be scheduled the task must past validation checks.
+		/// * The transaction is signed
+		/// * The provided_id's length > 0
+		/// * The time is valid
+		/// * 
+		///
+		/// # Parameters
+		/// * `provided_id`: An id provided by the user. This id must be unique for the user.
+		/// * `time`: The unix standard time in seconds for when the task should run.
+		/// * `message`: The message you want the event to have.
+		///
+		/// # Errors
+		/// * `InvalidTime`: Time must end in a whole minute.
+		/// * `PastTime`: Time must be in the future.
+		/// * `DuplicateTask`: There can be no duplicate tasks.
+		/// * `TimeSlotFull`: Time slot is full. No more tasks can be scheduled for this time.
+		/// * `InvalidAmount`: Amount has to be larger than 0.1 OAK.
+		/// * `InvalidRecipient`: Recipient must be a valid address.
+		/// * `TransferToSelf`: Sender cannot transfer money to self.
+		#[pallet::weight(<T as Config>::WeightInfo::schedule_notify_task_existing_slot())]
+		pub fn schedule_transfer_task(
+			origin: OriginFor<T>,
+			provided_id: Vec<u8>,
+			time: UnixTime,
+			recipient_id: T:: AccountId,
+			amount: u128, // TODO: jzhou fix to be type runtime::Balance
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let recipientExists = account_exists(recipient_id);
+			
+			// check for greater than existential deposit
+			ensure!(amount > EXISTENTIAL_DEPOSIT, Err(<Error<T>>::InvalidAmount))?;
```

Comment:
Ensure returns a result. The first param is the condition, the second is the error that is returned in the condition fails. Therefore you want it at the end of the ensure.

The ? says to pass the error up IF the fcn call it's after returns a result AND the fcn it is in returns a result

Reply:
yah, hm that was my thought as well, but when i put it after the ensure, i got this error: 
```the `?` operator can only be applied to values that implement `Try`
the trait `Try` is not implemented for `()`
```
it was much happier when i put it right after the error rather than outside ensure. although i prob need to test it cause it doesn't quite seem right